src/streambundle.ts

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright 2016 Teppo Kurki <teppo.kurki@iki.fi>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  StreamBundle as IStreamBundle,
```

Comment:
I debated about moving this entire implementation into `server-api` to avoid declaring an interface there and then implementing it here. The only other dependency here is `baconjs`, which is already a dep on `server-api`.